### PR TITLE
samples: basic: add multitarget_hello_world

### PR DESCRIPTION
**Pull request by gmarull**
Add a new "hello world" sample that can be built for multiple board
targets. This sample can be used to see if Zephyr boots successfully
with _minimal_ configuration on SoCs with multiple cores, or boards with
multiple SoCs.

The following configurations are provided for now:

- nRF5340 DK: cpuapp+cpunet
- nRF54H20 DK: cpuapp+cpurad, cpuapp+cpuppr, cpuapp+cpuppr(xip)

Note that because of a quoting bug somewhere in the build system, SB_CONFIG_REMOTE_BOARD is specified in .conf files (see the sysbuild/ directory). This should be fixed in the near future.

### CODE REVIEW

File: samples/basic/multicore_hello_world/sample.yaml
Comment (nordicjm) [inline samples/basic/multicore_hello_world/sample.yaml:32]: Needs porting to hwmv2

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: samples/basic/multicore_hello_world/sample.yaml
Comment (gmarull) [reply samples/basic/multicore_hello_world/sample.yaml:32]: yup, will rebase after hwmv2 is merged

File: samples/basic/multitarget_hello_world/sample.yaml
Comment (57300) [inline samples/basic/multitarget_hello_world/sample.yaml:2]: Rename?
```suggestion
  name: Multitarget Hello World
```

File: samples/basic/multitarget_hello_world/sample.yaml
Comment (57300) [inline samples/basic/multitarget_hello_world/sample.yaml:5]:
```suggestion
    Both images print the board target they were run on.
```

File: samples/basic/multitarget_hello_world/sysbuild.cmake
Comment (57300) [inline samples/basic/multitarget_hello_world/sysbuild.cmake:15]: Is it important for `remote` to be built and flashed first?

**Review by 57300 — COMMENTED** [review]
LGTM, although I wonder if it would be better to add a multitarget option to regular `hello_world` instead.

**Comment by gmarull** [comment]
@nordicjm ping

**Review by nordicjm — DISMISSED** [review]

**Review by fabiobaltieri — DISMISSED** [review]

**Comment by fabiobaltieri** [comment]
It's very tempting to suggest that this should have a multitarget blinky mode as well

File: samples/basic/multitarget_hello_world/README.rst
Comment (kartben) [inline samples/basic/multitarget_hello_world/README.rst:4]: I am a bit torn as to whether that should be a basic sample, and I don't think advertising it using "Hello World" terminology is effective (think: what would people use as keywords to find this)

I would suggest to move this to application_development and, while of course the code itself can be a hello world, I think the name & description should be more descriptive of the use case (similar to a comment also made by @57300).

Something like the below:

```suggestion
   :name: Multi-target application

   Use sysbuild to build and run on multiple board targets
```

**Review by kartben — CHANGES_REQUESTED** [review]

File: samples/basic/multitarget_hello_world/README.rst
Comment (gmarull) [reply samples/basic/multitarget_hello_world/README.rst:4]: I'm fine moving the sample. Shouldn't we have `samples/sysbuild`? Then we could have it under `samples/sysbuild/hello_world`, and call it "Multi-target hello world application".

Comment (kartben) [reply samples/basic/multitarget_hello_world/README.rst:4]: We already have `samples/application_development/sysbuild/with_mcuboot` so maybe `samples/application_development/sysbuild/hello_world` and add an index + ToC in `samples/application_development/sysbuild/` folder so that it's more visible? Or also fine to move sysbuild as a top-level folder but my point is the new sample should probably live next to the existing sysbuild mcuboot one.
Thanks!

Comment (carlescufi) [reply samples/basic/multitarget_hello_world/README.rst:4]: No objections from me, both options make sense.

Comment (gmarull) [reply samples/basic/multitarget_hello_world/README.rst:4]: `samples/application_development/sysbuild/with_mcuboot` is an extremely long path, I'd suggest shortening things. `application_development` is also very generic...

something like
- `samples/sysbuild/mcuboot`
- `samples/sysbuild/hello_world`

feels better IMHO.

Comment (fabiobaltieri) [reply samples/basic/multitarget_hello_world/README.rst:4]: ok since you are rescoping this, how about making it blink led-0 and led-1 (at different rates) as well? just to give some confidence that the cores are alive and running for boards where the integrated debugger only has a single uart available

Comment (gmarull) [reply samples/basic/multitarget_hello_world/README.rst:4]: > ok since you are rescoping this, how about making it blink led-0 and led-1 (at different rates) as well? just to give some confidence that the cores are alive and running for boards where the integrated debugger only has a single uart available

sounds like a good sample, but shouldn't it be ab extra `samples/sysbuild/blinky`, following our basic samples?

**Comment by gmarull** [comment]
last push: created a new category `samples/sysbuild` for samples that its main purpose is to demonstrate sysbuild features.

File: samples/basic/multitarget_hello_world/README.rst
Comment (gmarull) [reply samples/basic/multitarget_hello_world/README.rst]: I'll propose a samples/sysbuild/blinky if this moves on, not before, though.

**Review by carlescufi — DISMISSED** [review]

**Comment by carlescufi** [comment]
@kartben please take another look

**Review by MaureenHelm — DISMISSED** [review]

**Review by fabiobaltieri — DISMISSED** [review]

File: samples/sysbuild/with_mcuboot/README.rst
Comment (kartben) [inline samples/sysbuild/with_mcuboot/README.rst:1]: I'm started to think we should find a way to automate this somehow as I feel like a PITA always asking for this but: could you please make sure this has a redirect?
+1 otherwise, very cool stuff, thanks!

**Review by kartben — CHANGES_REQUESTED** [review]

File: samples/sysbuild/with_mcuboot/README.rst
Comment (gmarull) [reply samples/sysbuild/with_mcuboot/README.rst:1]: done

**Review by kartben — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Pull request merged**
